File: JavaScript/7js 1/getSum.js
/*
|--------------------------------------------------------------------------
| Beginner Series #3 Sum of Numbers
|--------------------------------------------------------------------------
|
*/

function getSum(a, b) {
   const arr = [];
   if (a > b)[a, b] = [b, a];
   for (let i = a; i <= b; i++) {
      arr.push(i)
   }
   return arr.reduce((x, y) => x + y);
}


function getSum(a, b) {
   let min = Math.min(a, b);
   let max = Math.max(a, b);
   return (max - min + 1) * (min + max) / 2;
}


function getSum(a, b) {
   return (a + b) * (Math.abs(a - b) + 1) / 2;
}


console.log(getSum(0, -1), -1);
console.log(getSum(0, 1), 1);
console.log(getSum(2, 6), 20);

File: Python/7py 2/get_sum.py
"""
|--------------------------------------------------------------------------
| Beginner Series #3 Sum of Numbers
|--------------------------------------------------------------------------
|
"""

def get_sum(a,b):
   minn = min(a,b)
   maxx = max(a,b)
   return (maxx - minn + 1) * (minn + maxx) / 2


def get_sum(a,b):
   return sum(range(min(a, b), max(a, b) + 1))


def get_sum(a, b):
   return (a + b) * (abs(a - b) + 1) // 2


def get_sum(a,b):
   a,b = sorted((a,b))
   return b*(b+1)//2 - a*(a-1)//2


print(get_sum(0,1),1)
print(get_sum(0,-1),-1)